connectors/cli/connector.py

```diff
@@ -0,0 +1,198 @@
+
+    async def list_connectors(self):
+        # TODO move this on top
+        try:
+            await self.es_client.ensure_exists(
+                indices=[CONCRETE_CONNECTORS_INDEX, CONCRETE_JOBS_INDEX]
+            )
+
+            connectors = []
+            async for connector in self.connector_index.all_connectors():
+                connectors.append(connector)
+
+            return connectors
+
+        # TODO catch exceptions
+        finally:
+            await self.connector_index.close()
+            await self.es_client.close()
+
+    async def ping(self):
+        if await self.es_client.ping():
+            await self.es_client.close()
+            return True
+        else:
+            return False
+
+    def service_type_configuration(self, source_class):
+        source_klass = get_source_klass(source_class)
+        configuration = source_klass.get_default_configuration()
+
+        return OrderedDict(sorted(configuration.items(), key=lambda x: x[1]["order"]))
+
+    def create(
+        self, index_name, service_type, configuration, language=DEFAULT_LANGUAGE
+    ):
+        return asyncio.run(
+            self.__create(index_name, service_type, configuration, language)
+        )
+
+    async def __create(
+        self, index_name, service_type, configuration, language=DEFAULT_LANGUAGE
+    ):
+        try:
+            return await asyncio.gather(
+                self.__create_search_index(index_name, language),
+                self.__create_connector(
+                    index_name, service_type, configuration, language
+                ),
+            )
+        except Exception as e:
+            raise e
+        finally:
+            await self.es_client.close()
+
+    async def __create_search_index(self, index_name, language):
+        mappings = Mappings.default_text_fields_mappings(
+            is_connectors_index=True,
+        )
+
+        settings = Settings(language_code=language, analysis_icu=False).to_hash()
+
+        settings["auto_expand_replicas"] = "0-3"
+        settings["number_of_shards"] = 2
+
+        try:
+            await self.es_client.client.indices.create(
+                index=index_name, mappings=mappings, settings=settings
+            )
+        except Exception as e:
+            # todo handle exception
+            raise e
+
+    async def __create_connector(
+        self, index_name, service_type, configuration, language
+    ):
+        try:
+            await self.es_client.ensure_exists(
+                indices=[CONCRETE_CONNECTORS_INDEX, CONCRETE_JOBS_INDEX]
+            )
+            timestamp = iso_utc()
+
+            doc = {
+                "api_key_id": "",
+                "configuration": configuration,
+                "index_name": index_name,
+                "service_type": service_type,
+                "status": "configured",  # TODO use a predefined constant
+                "is_native": True,  # TODO make it optional
```

Comment:
Yes, this CLI should be able to create both native/non-native connectors. 

> If we do support creating native connectors in the CLI, I think is_native should maybe be False by default

In this PR only native connectors are supported. Non-native connectors support requires API key generation and probably config.yml modification. I will add non-native support soon because I need them for some automation.